lib/features/family/features/login/presentation/pages/family_login_sheet.dart

```diff
@@ -2,72 +2,71 @@ import 'package:flutter/material.dart';
+import 'package:givt_app/features/family/shared/widgets/loading/custom_progress_indicator.dart';
 import 'package:givt_app/features/family/shared/widgets/texts/shared_texts.dart';
 import 'package:givt_app/l10n/l10n.dart';
 import 'package:givt_app/shared/dialogs/dialogs.dart';
 import 'package:givt_app/shared/models/analytics_event.dart';
+import 'package:givt_app/shared/widgets/base/base_state_consumer.dart';
 import 'package:givt_app/shared/widgets/outlined_text_form_field.dart';
 import 'package:givt_app/utils/util.dart';
 import 'package:go_router/go_router.dart';
 
-class FamilyLoginPage extends StatefulWidget {
-  const FamilyLoginPage({
-    required this.email,
-    this.navigate,
+class FamilyLoginSheet extends StatefulWidget {
+  const FamilyLoginSheet({
+    required this.navigate,
+    this.email,
     super.key,
   });
 
-  final String email;
-  final Future<void> Function(BuildContext context, {bool? isUSUser})? navigate;
+  final String? email;
+  final Future<void> Function(BuildContext context) navigate;
 
   @override
-  State<FamilyLoginPage> createState() => _FamilyLoginPageState();
+  State<FamilyLoginSheet> createState() => _FamilyLoginSheetState();
 }
 
-class _FamilyLoginPageState extends State<FamilyLoginPage> {
+class _FamilyLoginSheetState extends State<FamilyLoginSheet> {
   final formKey = GlobalKey<FormState>();
   late TextEditingController emailController;
   late TextEditingController passwordController;
   bool obscureText = true;
 
+  final _cubit = getIt<FamilyLoginCubit>();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    _cubit.init(widget.email);
+  }
+
   @override
   void initState() {
     super.initState();
-    emailController = TextEditingController(text: widget.email);
+    emailController = TextEditingController();
     passwordController = TextEditingController();
   }
 
   Future<void> onLogin(BuildContext context) async {
-    if (formKey.currentState!.validate()) {
-      try {
-        await context
-            .read<AuthCubit>()
-            .login(
-              email: emailController.text,
-              password: passwordController.text,
-              navigate: widget.navigate,
-            )
-            .whenComplete(() {
-          if (!context.mounted) return;
-          final authState = context.read<AuthCubit>().state;
-
-          if (authState.status == AuthStatus.authenticated ||
-              authState.status == AuthStatus.biometricCheck) {
-            context.pop(true);
-          }
-        });
-      } catch (e) {
-        if (!context.mounted) return;
-
-        ScaffoldMessenger.of(context).showSnackBar(
-          SnackBar(
-            content: Text(e.toString()),
-          ),
-        );
-      }
+    if (!formKey.currentState!.validate()) return;
+
+    final loginResult = await _cubit.login(
+      emailController.text,
+      passwordController.text,
+    );
+
+    if (loginResult) {
```

Comment:
I would not do this here and I would not let the login method return a boolean. The cubit should handle what happens afterwards. I would add a custom navigate event instead. So when login succeeds the cubit emits the custom navigate event. The view just handles the custom events but is "unaware" of login logic.